src/language/term/Typ.re

```diff
@@ -133,6 +133,33 @@ let rec is_arrow = (typ: t) => {
   };
 };
 
+let rec has_arrow = (ctx: Ctx.t, typ: t) =>
+  switch (typ.term) {
+  | Parens(typ) => has_arrow(ctx, typ)
+  | TupLabel(_, typ) => has_arrow(ctx, typ)
+  | Arrow(_)
+  | Forall(_) => true
+  | Unknown(_)
+  | Atom(_)
+  | Label(_) => false
+  | Var(tvar) =>
+    switch (Ctx.lookup_alias(ctx, tvar)) {
```

Comment:
normalize before calling this function so you don't have to handle these things -- then you don't need a context